data/state/accountsDB.go

```diff
@@ -123,18 +134,48 @@ func (adb *AccountsDB) LoadDataTrie(accountHandler AccountHandler) error {
 	return nil
 }
 
+// searches a data entry in journal from the most recent entries towards the first
+// and if it finds a matching data trie, returns a deep copy of that trie
+func (adb *AccountsDB) searchDataTrieInJournal(rootHash []byte) (data.Trie, error) {
+	adb.mutEntries.RLock()
+	defer adb.mutEntries.RUnlock()
+
+	for i := len(adb.entries) - 1; i >= 0; i-- {
+		dataTrieEntry, ok := adb.entries[i].(*BaseJournalEntryData)
+		if !ok {
+			continue
+		}
+
+		dataRootHash, err := dataTrieEntry.trie.Root()
+		if err != nil {
+			return nil, errors.New("corrupted trie found in journal while computing root hash: " + err.Error())
+		}
+
+		if bytes.Equal(dataRootHash, rootHash) {
+			return dataTrieEntry.trie.DeepClone()
+		}
+	}
+
+	return nil, NewErrMissingTrie(rootHash)
```

Comment:
if missing you could log.Debug the missing roothash, and return a constant error.

Reply:
We should shift from reusing the same error vars (that do not explain too much what is wrong) to more comprehensible error types (non constant errors). Again, this helps the debugging process and it is go-idiomatic.